osquery/tables/system/windows/processes.cpp

```diff
@@ -148,238 +216,344 @@ Status getProcList(std::set<long>& pids) {
-    std::vector<char> tokUser(sizeof(TOKEN_USER), 0x0);
-    auto ret = OpenProcessToken(hProcess, TOKEN_READ, &tok);
-    if (ret != 0 && tok != nullptr) {
-      unsigned long tokOwnerBuffLen;
-      ret = GetTokenInformation(tok, TokenUser, nullptr, 0, &tokOwnerBuffLen);
-      if (ret == 0 && GetLastError() == ERROR_INSUFFICIENT_BUFFER) {
-        tokUser.resize(tokOwnerBuffLen);
-        ret = GetTokenInformation(
-            tok, TokenUser, tokUser.data(), tokOwnerBuffLen, &tokOwnerBuffLen);
-      }
+  unsigned long size_out = 0;
+  PROCESS_BASIC_INFORMATION proc_info;
+  SecureZeroMemory(&proc_info, sizeof(PROCESS_BASIC_INFORMATION));
 
-      // Check if the process is using an elevated token
-      auto elevated = FALSE;
-      TOKEN_ELEVATION Elevation;
-      DWORD cbSize = sizeof(TOKEN_ELEVATION);
-      if (GetTokenInformation(
-              tok, TokenElevation, &Elevation, sizeof(Elevation), &cbSize)) {
-        elevated = Elevation.TokenIsElevated;
-      }
+  // See the Process Hacker implementation for more details on the hard coded 60
+  // https://github.com/processhacker/processhacker/blob/master/phnt/include/ntpsapi.h#L160
+  auto ret = kNtQueryInformationProcess(proc, 60, NULL, 0, &size_out);
 
-      r["is_elevated_token"] = elevated ? INTEGER(1) : INTEGER(0);
-    }
-    if (uid != 0 && ret != 0 && !tokUser.empty()) {
-      auto sid = PTOKEN_OWNER(tokUser.data())->Owner;
-      r["uid"] = INTEGER(getUidFromSid(sid));
-      r["gid"] = INTEGER(getGidFromSid(sid));
-    } else {
-      r["uid"] = INTEGER(uid);
-      r["gid"] = INTEGER(gid);
-    }
-    if (tok != nullptr) {
-      CloseHandle(tok);
-      tok = nullptr;
-    }
+  if (ret != STATUS_BUFFER_OVERFLOW && ret != STATUS_BUFFER_TOO_SMALL &&
+      ret != STATUS_INFO_LENGTH_MISMATCH) {
+    return Status(1,
+                  "NtQueryInformationProcess failed for " +
+                      std::to_string(pid) + " with " + std::to_string(ret));
   }
 
-  if (hProcess != nullptr) {
-    CloseHandle(hProcess);
+  auto cmdline = static_cast<PVOID>(malloc(size_out));
+  ret = kNtQueryInformationProcess(proc, 60, cmdline, size_out, &size_out);
+
+  if (!NT_SUCCESS(ret)) {
+	  free(cmdline);
+    return Status(1,
+                  "NtQueryInformationProcess failed for " +
+                      std::to_string(pid) + " with " + std::to_string(ret));
   }
+  auto ustr = reinterpret_cast<PUNICODE_STRING>(cmdline);
+  out = wstringToString(ustr->Buffer);
+  free(cmdline);
+  return Status();
 }
 
-// collect perf data into a hashmap by pid to later be refferenced
+void getProcessPathInfo(HANDLE& proc, const unsigned long pid, DynamicTableRowHolder& r) {
+  auto out = kMaxPathSize;
+  std::unique_ptr<char> path(new char[kMaxPathSize]);
+  SecureZeroMemory(path.get(), kMaxPathSize);
+  auto ret = QueryFullProcessImageName(proc, 0, path.get(), &out);
+  if (ret != TRUE) {
+    VLOG(1) << "Failed to lookup path information for process " << pid;
+    r["path"] = "-1";
```

Comment:
setting path to "-1" on failure may lead to weirdness.  For example when joining on processes table to do something, at least one process will have path='-1' when running as daemon with SYSTEM privs.  The following query will log WARN status `authenticode.cpp:488] Failed to verify the Authenticode signature for the following file: -1.`.
```
SELECT path, serial_number, result FROM authenticode WHERE path IN (SELECT path FROM processes GROUP BY path)
```

Reply:
Definitely. Do you have a preferred alternative? We can leave the value blank as well. We've commonly used `-1` to indicate some failure state of osquery to get the information it's seeking.

Reply:
I leave it blank.